app/models/state_changers.rb

```diff
@@ -111,81 +171,46 @@ def labware
   end
 
   # The tube state changer is used by Tubes. It works the same way as the default
-  # state changer but does not need to handle a subset of wells like the plate.
-  class TubeStateChanger < DefaultStateChanger
+  # plate state changer but does not need to handle a subset of wells like the plate.
+  class TubeStateChanger < BaseStateChanger
     # Tubes have no wells so contents is always empty
     def contents_for(_target_state)
       nil
     end
-  end
-
-  # Plate state changer to automatically complete specified work requests.
-  # This is the abstract version.
-  class AutomaticLabwareStateChanger < DefaultStateChanger
-    def v2_labware
-      raise 'Must be implemented on subclass'
-    end
-
-    def purpose_uuid
-      @purpose_uuid ||= v2_labware.purpose.uuid
-    end
-
-    def purpose_config
-      @purpose_config ||= Settings.purposes[purpose_uuid]
-    end
-
-    def work_completion_request_types
-      @work_completion_request_types ||= parse_work_completion_request_types
-    end
-
-    # config can be a single request type or an array of request types
-    # convert them here into a consistent array format
-    def parse_work_completion_request_types
-      config = purpose_config[:work_completion_request_type]
-      return config if config.is_a?(Array)
-
-      [config]
-    end
-
-    # rubocop:todo Style/OptionalBooleanParameter
-    def move_to!(state, reason = nil, customer_accepts_responsibility = false)
-      super
-      complete_outstanding_requests
-    end
 
-    # rubocop:enable Style/OptionalBooleanParameter
-
-    def complete_outstanding_requests
-      in_prog_submissions =
-        v2_labware.in_progress_submission_uuids(request_types_to_complete: work_completion_request_types)
-      return if in_prog_submissions.blank?
-
-      api.work_completion.create!(submissions: in_prog_submissions, target: v2_labware.uuid, user: user_uuid)
+    def labware
+      raise 'Tubes are not supported by API V1'
     end
   end
 
   # This version of the AutomaticLabwareStateChanger is used by Plates.
-  class AutomaticPlateStateChanger < AutomaticLabwareStateChanger
+  class AutomaticPlateStateChanger < PlateStateChanger
+    include AutomaticBehaviour
+
     def v2_labware
       @v2_labware ||= Sequencescape::Api::V2.plate_for_completion(labware_uuid)
     end
   end
 
-  # This version of the AutomaticLabwareStateChanger is used by Tubes.
-  class AutomaticTubeStateChanger < AutomaticLabwareStateChanger
+  # This version of the AutomaticLabwareStateChanger is used by TubeRacks.
+  class AutomaticTubeRackStateChanger < TubeRackStateChanger
+    include AutomaticBehaviour
+
     def v2_labware
-      @v2_labware ||= Sequencescape::Api::V2.tube_for_completion(labware_uuid)
+      @v2_labware ||= Sequencescape::Api::V2.tube_rack_for_completion(labware_uuid)
     end
 
     def labware
```

Comment:
this labware vs v2_labware nonsense will hopefully get refactored out in a later story!